### DISCUSSION TITLE
Connect to WiFi - allow static IP address configuration

### DISCUSSION THREAD

**Discussion by ghost** [OnionUI/Onion#1089] [Ideas]
Hello there. :wave:
First of all, thank you all for such amazing work!

I'd like to ask if you're considering adding the option to configure static IP addresses for any given connection.

Doing a non-representative, shallow research (basically `grep`ing inside the `.tmp_update` folder for `udhcpc`), I see multiple references to the DHCP service, scattered across many files, so I understand it can be non-trivial to implement such functionality.

Additionally, the structure of such configuration should also be discussed (One config for any connection? Store configs for each connection?).

Finally, I'm aware connecting to a WiFi spot should be as easy as possible (hence the use of DHCP).

Thank you in advance for your consideration, and keep up the good work! :+1:

**Comment by idxman01** [comment]
I would add a dhcp reservation in your fw/router.

**Reply by XK9274** [reply]
> I would add a dhcp reservation in your fw/router.

The MMP does support statics and configuring statics but only currently through terminal, there's no front end for it so the above answer is the best solution currently.

**Reply by ghost** [reply]
> > I would add a dhcp reservation in your fw/router.
> 
> The MMP does support statics and configuring statics but only currently through terminal, there's no front end for it so the above answer is the best solution currently.

Thank you for your responses!

I'm actually open to doing it via terminal commands.

I can't do it via router configuration, because my ISP is so protective I'm not even allowed to change simple settings like turning DHCP off, or even daisy-chaining a custom router, without losing the connection.

I was just cautious with the tight integration I saw of `udhcpc`, but if you say it can be done, any insight is appreciated. :pray:

**Reply by XK9274** [reply]
Got busy, if you want you can setup an app/script that will set your IP/mask/gateway using the below (obv change the ip details)

```
    ifconfig wlan0 up
    killall -9 udhcpc
    ip addr flush dev wlan0
    ip addr add 192.168.100.101/24 dev wlan0
    ip route add default via 192.168.100.100
```

which is how i handle it in the pokemon scripts, however it won't work over reboots unless you add it to your runtime.sh or use another method of running it at boot (and then it'll only work if wlan0 is actually available, i.e wifi is enabled when the script is run) - honestly it's easier to do it through your router, crazy your ISP doesn't let you do anything with the router.
